libsolidity/formal/Predicate.cpp

```diff
@@ -196,12 +198,20 @@ bool Predicate::isInterface() const
 	return m_type == PredicateType::Interface;
 }
 
-string Predicate::formatSummaryCall(vector<smtutil::Expression> const& _args) const
+string Predicate::formatSummaryCall(
+	vector<smtutil::Expression> const& _args,
+	langutil::CharStreamBySourceName const& _charStreamProvider
+) const
 {
 	solAssert(isSummary(), "");
 
 	if (auto funCall = programFunctionCall())
-		return funCall->location().text();
+	{
+		if (funCall->location().hasText())
+			return string(_charStreamProvider.charStream(*funCall->location().sourceName).text(funCall->location()));
+		else
+			return {};
```

Comment:
If you're passing the location into `text()` anyway, why not put the `if` inside the helper too? You could simply return `string_view{}` when there's no text.

Reply:
With the `if` moved into the helper this can be now simplified:
```suggestion
		string(_charStreamProvider.charStream(*funCall->location().sourceName).text(funCall->location()));
```
Or is there a problem with converting an empty `string_view` to `string`?

Reply:
ok, I see the problem. https://github.com/ethereum/solidity/pull/11596#discussion_r669597731